### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'to'

### DISCUSSION THREAD

**Discussion by hotarugali** [zai-org/ChatGLM3#1221] [Q&A]
I split one dataset into `train.json` and 'valid.json', so their format are the same. When I finetune my chatglm3, it throws error `AttributeError: 'NoneType' object has no attribute 'to'` during evaltion. The training succeeded, but evaluating failed. 
It seems the different way to `get_dataset` in `finetune_hf.py`. The `process_batch_eval` doesn't generate `labels`, making there are `{'labels': None}` in the dataset.

![图片](https://github.com/THUDM/ChatGLM3/assets/32862896/bfca5047-d237-4fb7-879f-1f3131d0af36)

![图片](https://github.com/THUDM/ChatGLM3/assets/32862896/024a2b2b-aff7-45db-a966-08dc7058cb53)

![图片](https://github.com/THUDM/ChatGLM3/assets/32862896/0d4d5793-e7bc-4456-88e5-2042d935134f)

**Comment by duanhao123** [comment]
I have the same problem, I hope someone can answer it, 
If anyone answers it, let me know
thank you ! !

**Reply by QJ1234** [reply]
The guy above gave me an answer: try install transformers==4.40.0. I tried it and worked.Maybe you can have a try

**Comment by morettt** [comment]
我也遇到了这个问题，前面微调正常，后面到了保存点的时候就出现了这个报错。

**Reply by QLiangong** [reply]
朋友，想问一下你有解决方法了吗

**Comment by QLiangong** [comment]
我也遇到了同样的问题。TypeError: BatchEncoding.to() got an unexpected keyword argument 'non_blocking'，我的解决方法是将operations.py对应出错的位置上的non_blocking=non_blocking去除，我猜是tensor.to()这个函数并没有对应non_blocking的属性，所以无论true和false都没用。AttributeError: 'NoneType' object has no attribute 'to'，这个问题我猜是数据不知道为什么NoneType，所以.to()的时候没法使用，我也不知道该怎么解决，希望有人能帮忙

**Reply by QJ1234** [reply]
把transformers换成4.40.0，我这里不报错了

**Comment by morettt** [comment]
> 朋友，想问一下你有解决方法了吗

没有，还是一样。这个只能等官方维护了，或者用之前的历史版本。

**Comment by QJ1234** [comment]
同样的问题，一跑到eval-step的时候就报这个错，当我尝试修改eval-step大于max-step，再次运行时，在进行最后的prediction又报这个错。

**Reply by morettt** [reply]
> 同样的问题，一跑到eval-step的时候就报这个错，当我尝试修改eval-step大于max-step，再次运行时，在进行最后的prediction又报这个错。

你去试一下把transformers  换成4.40.0这个版本的。我昨天试了一下换成4.40 不会出现这个报错

**Reply by QJ1234** [reply]
测试有用，官方主文件夹中给的requirements.txt中写的是transformers>=4.41.0。可能微调和运行不太一样吧

**Reply by BLANKBLK** [reply]
但是换成4.40会直接报错，会出现没有phi3模块。怎么办？

**Comment by JasonYuyzy** [comment]
Got the same problem.
If anyone could solve it please let me know!
XXX

**Reply by QJ1234** [reply]
The guy above gave me an answer: try install transformers==4.40.0. I tried it and worked.Maybe you can have a try

**Reply by JasonYuyzy** [reply]
Cheers, I will try that. 谢谢

**Reply by JasonYuyzy** [reply]
![transformer_problems](https://github.com/THUDM/ChatGLM3/assets/44361310/d90e3dbd-850f-4bed-94b0-49150600d805)
但我在训练英文数据的时候出现了这个新问题

**Reply by zhoulq-789** [reply]
我也遇到了这个问题，请问是怎么解决的
AttributeError: 'NoneType' object has no attribute 'pop'

**Reply by QJ1234** [reply]
时间有点久了朋友，我记得当时可能是显存不够之类的设备原因，你可以试试用低显存的方案跑一跑

**Reply by JasonYuyzy** [reply]
尝试换不同的版本，后面无论llama3.1、3.2，ChatGLM3、4对应的transformers的版本都不一样，找回官方对应版本就好了

**Reply by JasonYuyzy** [reply]
还有一个能解决的就是取消validation步骤

**Comment by HXX2020** [comment]
transformer换成4.40.0亲测有用

**Reply by BLANKBLK** [reply]
transformer换成4.40.0后，会出现No module named 'transformers.models.phi3'的错误。怎么解决？

**Reply by Jayshon1** [reply]
是啊我也遇到这个问题你怎么解决的吗
